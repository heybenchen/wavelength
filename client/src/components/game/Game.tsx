import { Chip, makeStyles } from "@material-ui/core";
import React, { useEffect, useState } from "react";
import io from "socket.io-client";
import Device from "../device/Device";
import Score from "../score/Score";
import { useParams } from "react-router-dom";

const DEVELOPMENT_PORT = ":9001";

const useStyles = makeStyles({
  root: {
    textAlign: "center",
    height: "100%",
    width: "100%",
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    fontSize: "calc(10px + 1.5vmin)",
    backgroundColor: "#c2cdd4",
  },
  status: {
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    width: "100%",
    marginTop: "16px",
  },
});

function Game() {
  const classes = useStyles();
  const { roomId } = useParams();
  const [connectedClients, setConnectedClients] = useState([""]);
  const [socket, setSocket] = useState<SocketIOClient.Socket>();

  useEffect(() => {
    const isDevelopmentMode = process.env.NODE_ENV === "development";
    const socket = isDevelopmentMode ? io(DEVELOPMENT_PORT) : io();
    socket.on("connected ids", (data: Object) => setConnectedClients(Object.keys(data)));
    socket.on("connect", () => socket.emit("join room", roomId));
    setSocket(socket);

    return function cleanup() {
      socket.disconnect();
    };
  }, [roomId]);

  const getPlayersString = () => {
    const playerCount = Object.keys(connectedClients).length;
    if (playerCount <= 1) {
      return "Waiting for players";
    }
    return `${playerCount} Players`;
  };

  return (
    <div className={classes.root}>
      <div className={classes.status}>
        <Score socket={socket} teamId={0} />
        <Chip label={getPlayersString()} />
        <Score socket={socket} teamId={1} />
      </div>
      <Device socket={socket} />
    </div>
  );
}

export default Game;
