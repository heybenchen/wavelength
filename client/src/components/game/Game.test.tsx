import { render } from "@testing-library/react";
import React from "react";
import Game from "./Game";

test("renders player count", () => {
  const { getByText } = render(<Game />);
  const playerCountElement = getByText(/Players/i);
  expect(playerCountElement).toBeInTheDocument();
});
