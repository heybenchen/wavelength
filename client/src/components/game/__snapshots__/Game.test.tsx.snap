// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <div
          class="makeStyles-status-2"
        >
          <div
            class="makeStyles-teamContainer-3"
          >
            <div
              class="MuiButtonGroup-root makeStyles-root-5 MuiButtonGroup-contained"
              role="group"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedSecondary makeStyles-button-6 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  -
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedSecondary makeStyles-button-6 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  0
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedSecondary makeStyles-button-6 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  +
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="MuiButtonBase-root MuiChip-root MuiChip-clickable"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiChip-label"
            >
              Waiting for players
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
          <div
            class="makeStyles-teamContainer-3"
          >
            <div
              class="MuiButtonGroup-root makeStyles-root-5 MuiButtonGroup-contained"
              role="group"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-button-6 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  -
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-button-6 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  0
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-button-6 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  +
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-root-7"
        >
          <div
            class="makeStyles-root-17"
          >
            <div
              class="makeStyles-card-20"
              style="background-color: rgb(120, 180, 111); border-radius: 8px 0px 0px 8px;"
            >
              <div
                class="makeStyles-cardArrow-19"
              >
                ⟵
              </div>
              <div
                class="makeStyles-cardText-18"
              />
            </div>
            <div
              class="makeStyles-card-20"
              style="background-color: rgb(227, 181, 95); border-radius: 0px 8px 8px 0px;"
            >
              <div
                class="makeStyles-cardArrow-19"
              >
                ⟶
              </div>
              <div
                class="makeStyles-cardText-18"
              />
            </div>
          </div>
          <div
            class="makeStyles-deviceContainer-8"
          >
            <img
              alt="Device Answer"
              class="makeStyles-deviceImg-9 makeStyles-deviceAnswer-10"
              src="Target.svg"
              style="transform: rotate(0deg);"
            />
            <img
              alt="Device Visor"
              class="makeStyles-deviceImg-9 makeStyles-deviceVisor-13"
              src="Visor.svg"
              style="transform: rotate(0deg); transition: transform 2s ease, opacity 1s ease; opacity: 1;"
            />
            <img
              alt="Device Cover"
              class="makeStyles-deviceImg-9 makeStyles-deviceCover-11"
              draggable="false"
              src="Cover.svg"
            />
            <img
              alt="Device Dial"
              class="makeStyles-deviceImg-9 makeStyles-deviceDial-12"
              src="Dial.svg"
              style="transform: rotate(0deg);"
            />
          </div>
          <div
            class="makeStyles-buttonContainer-14"
          >
            <div
              class="makeStyles-spacer-16"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-15"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Peek
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div
              class="makeStyles-spacer-16"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-15"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Submit
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div
              class="makeStyles-spacer-16"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-15"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                New
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <div
              class="makeStyles-spacer-16"
            />
          </div>
        </div>
        <div />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <div
        class="makeStyles-status-2"
      >
        <div
          class="makeStyles-teamContainer-3"
        >
          <div
            class="MuiButtonGroup-root makeStyles-root-5 MuiButtonGroup-contained"
            role="group"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedSecondary makeStyles-button-6 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                -
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedSecondary makeStyles-button-6 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                0
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedSecondary makeStyles-button-6 MuiButton-containedSecondary MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                +
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiButtonBase-root MuiChip-root MuiChip-clickable"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiChip-label"
          >
            Waiting for players
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
        <div
          class="makeStyles-teamContainer-3"
        >
          <div
            class="MuiButtonGroup-root makeStyles-root-5 MuiButtonGroup-contained"
            role="group"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-button-6 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                -
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-button-6 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                0
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButtonGroup-grouped MuiButtonGroup-groupedHorizontal MuiButtonGroup-groupedContained MuiButtonGroup-groupedContainedHorizontal MuiButtonGroup-groupedContainedPrimary makeStyles-button-6 MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                +
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-root-7"
      >
        <div
          class="makeStyles-root-17"
        >
          <div
            class="makeStyles-card-20"
            style="background-color: rgb(120, 180, 111); border-radius: 8px 0px 0px 8px;"
          >
            <div
              class="makeStyles-cardArrow-19"
            >
              ⟵
            </div>
            <div
              class="makeStyles-cardText-18"
            />
          </div>
          <div
            class="makeStyles-card-20"
            style="background-color: rgb(227, 181, 95); border-radius: 0px 8px 8px 0px;"
          >
            <div
              class="makeStyles-cardArrow-19"
            >
              ⟶
            </div>
            <div
              class="makeStyles-cardText-18"
            />
          </div>
        </div>
        <div
          class="makeStyles-deviceContainer-8"
        >
          <img
            alt="Device Answer"
            class="makeStyles-deviceImg-9 makeStyles-deviceAnswer-10"
            src="Target.svg"
            style="transform: rotate(0deg);"
          />
          <img
            alt="Device Visor"
            class="makeStyles-deviceImg-9 makeStyles-deviceVisor-13"
            src="Visor.svg"
            style="transform: rotate(0deg); transition: transform 2s ease, opacity 1s ease; opacity: 1;"
          />
          <img
            alt="Device Cover"
            class="makeStyles-deviceImg-9 makeStyles-deviceCover-11"
            draggable="false"
            src="Cover.svg"
          />
          <img
            alt="Device Dial"
            class="makeStyles-deviceImg-9 makeStyles-deviceDial-12"
            src="Dial.svg"
            style="transform: rotate(0deg);"
          />
        </div>
        <div
          class="makeStyles-buttonContainer-14"
        >
          <div
            class="makeStyles-spacer-16"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-15"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Peek
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="makeStyles-spacer-16"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-15"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="makeStyles-spacer-16"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-15"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              New
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            class="makeStyles-spacer-16"
          />
        </div>
      </div>
      <div />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
